src/lib/tabs/tab-header.ts

```diff
@@ -0,0 +1,258 @@
+import {MdTabLabelWrapper} from './tab-label-wrapper';
+import {MdInkBar} from './ink-bar';
+import 'rxjs/add/operator/map';
+
+export type ScrollDirection = 'after' | 'before';
+
+@Component({
+  moduleId: module.id,
+  selector: 'md-tab-header',
+  templateUrl: 'tab-header.html',
+  styleUrls: ['tab-header.css'],
+  encapsulation: ViewEncapsulation.None,
+  host: {
+    '[class.md-tab-header]': 'true',
+    '[class.scroll]': '_isScrollingEnabled()',
+    '[class.ltr]': "_getLayoutDirection() == 'ltr'",
+    '[class.rtl]': "_getLayoutDirection() == 'rtl'",
+  }
+})
+export class MdTabHeader {
+  @ContentChildren(MdTabLabelWrapper) _labelWrappers: QueryList<MdTabLabelWrapper>;
+
+  @ViewChild(MdInkBar) _inkBar: MdInkBar;
+  @ViewChild('tabListContainer') _tabListContainer: ElementRef;
+  @ViewChild('tabList') _tabList: ElementRef;
+
+  /** The tab index that is focused. */
+  private _focusIndex: number = 0;
+
+  /** Index to focus after the view has been checked. */
+  private _indexToFocus: number;
+
+  /** The distance in pixels that the tab labels should be translated to the left. */
+  private _scrollDistance = 0;
+
+  /** Whether the header should scroll to the selected index after the view has been checked. */
+  private _scrollToSelectedIndex = false;
+
+  private _disableScrollAfter = true;
+  private _disableScrollBefore = true;
+
+  /** The index of the active tab. */
+  private _selectedIndex: number = 0;
+  @Input() set selectedIndex(value: number) {
+    this._selectedIndex = value;
+    this._focusIndex = value;
+
+    // Scroll to this index on the next view check in case the tab has not yet been created.
+    this._scrollToSelectedIndex = true;
+  }
+  get selectedIndex(): number {
+    return this._selectedIndex;
+  }
+
+  /** Event emitted when the option is selected. */
+  @Output() selectFocusedIndex = new EventEmitter();
+
+  /** Event emitted when a label is focused. */
+  @Output() indexFocused = new EventEmitter();
+
+  constructor(private _zone: NgZone,
+              private _elementRef: ElementRef,
+              private _renderer: Renderer,
+              @Optional() private _dir: Dir) {}
+
+  /**
+   * Waits one frame for the view to update, then updates the ink bar and scroll.
+   * Note: This must be run outside of the zone or it will create an infinite change detection loop
+   * TODO: internal
+   */
+  ngAfterViewChecked(): void {
+    this._zone.runOutsideAngular(() => {
+      window.requestAnimationFrame(() => {
+        this._updateInkBar();
+        this._updateScrollPosition();
+
+        this._disableScrollBefore = this.scrollDistance == 0;
+        this._disableScrollAfter = this.scrollDistance == this._getMaxScrollDistance();
+      });
+    });
+
+    if (this._scrollToSelectedIndex) {
+      if (this._isScrollingEnabled()) {
+        this._scrollToLabel(this._selectedIndex);
+      }
+      this._scrollToSelectedIndex = false;
+    }
```

Comment:
Can you break this logic into a separate method like `_updateTabScrollPosition` or similar? Then just call the method in ngAfterViewChecked